embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/resurrection/PayloadResurrectionServiceImpl.kt

```diff
@@ -35,18 +36,25 @@ internal class PayloadResurrectionServiceImpl(
 ) : PayloadResurrectionService {
 
     override fun resurrectOldPayloads() {
+        val nativeCrashService = nativeCrashServiceProvider()
+        val nativeCrashes = nativeCrashService?.getNativeCrashes()?.associateBy { it.sessionId } ?: emptyMap()
```

Comment:
Perhaps a future refactor could move this to within the `NativeCrashService`, assuming nothing else needs to get a list of native crashes?

Reply:
Yeah definitely. The NdkService-related classes and interfaces seem like they could use some help. Like, some of this stuff, I think I should jus be pulling from the NDK directly and bypass this service. In fact, the service seems like it should only be sending the crash logs and not doing anything else.

All this can be consolidated once we get rid of v1. I'll add a task